### PR TITLE
Add Ṣifhu Lī

### PR DESCRIPTION
**Pull request by adehad**

### CODE REVIEW

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+            ["صفْ جَمِيلًا أَكْحَلًا مِنْ غَيْرِ كُحْلٍ"],
+            ["أَدْعَجًا عَيْنَاهُ تُنْسِيكَ الغَزَالَا"],
+
+            ["صِفْ غَضُوضَ الطَّرفِ بَسَّامُ المُحَيَّا "],
+            ["أَنْجَلًا تُنْسِيكَ طَلْعَتُهُ الهِلَالَا"],
+
+            ["صِفْ جَمِيلَ القَدِّ وَرْدِيَّ الوِجَانَا "],
+            ["صِفْ مَلِيحًا حُسْنُهُ فَاقَ الخَيَالًا"],
+
+            ["صِفْ بِهِ عُنُقًا مُنِيرًا كَوْكَبِِيًّا"],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:32]: ["صِف بِهِ عُنُقاً مُنِيراً كَوكَبِياً"] ,

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+            ["صِفْ جَمِيْلًا نُورُهُ فِي الكَوْنِِ لَالَا"],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:33]: ["صِفْ جَمِيْلًا نُورُهُ فِي الكَونِ لَالَا"],

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+            ["صِفْ ضَلِيعَ الفَمِ بَرَّاقَ الثَّنَايَا"],
+            ["صِفْ نَدِيَّ الرَّاحِ مِنْهُ الغَيْتُ سَالَا"],
+
+            ["صِفْ طَوِيلَ الهُدْبِ صِفْ أَنْفًا كَسَيْفٍ "],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:38]: ["صِفْ طَوِيلَ الهُدْبِ صِفْ أنْفَاً كَسَيْفٍ "],

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+            ["صِفْ أَزَجَّ الحَاجِبَيْنِِ بِلَا اتِّصالَا"],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:39]: ["صِفْ أَزَجَّ الحَاجِبَيْنِ بِلَا اتِّصالَاً"],

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+            ["صِفْ نَبِيًّا قَدْ أَتَى مِنْ قَبْلِ آدمْ "],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:41]: ["صِفْ نَبِيّاً قَدْ أَتَى مِنْ قَبْلِ آدَمْ "],

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+            ["جَلَّ مَنْ سَوَّاهُ لَيْسَ لَهُ مِثَالَا"],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:42]: ["جَلَّ مَنْ سَوَّاهُ لَيْسَ لَهُ مِثَالَاً"],

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+
+            ["أَبْصَرَتْ عَيْنَاكَ طَلْعَةَ مُصْطَفَانَا"],
+            ["قُلْ بِرَبِّكَ كَيْفَ أَدْرَكْتَ الوِصَالَا"],
+
+            ["كَمْ أُنَادِي يَا أَبَا الزَهْرَاءْ أَقْبِل"],
+            ["كَمْ أٌنادٍي يَا أَبَا الزَهْرَا تَعَالَ"],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:48]: ["كَمْ أُنَادِي يَا أَبَا الزَهرَاء تَعَالَ"],

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+            ["صِحْتُ وَاشَوْقِاه وَجْداً يَا حَبِيبِي"],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:50]: ["صِحْتُ وَاشَوْقَاه وَجْداً يَا حَبِيبِي"],

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 9]
```diff
@@ -0,0 +1,214 @@
+{
+    "name": {
+        "ara": "صفه لي",
+        "en": "Ṣifhu Lī – Describe him to me"
+    },
+    "verses": {
+        "ara": [
+            ["صِفْهُ لِي يَا مَنْ رَأَيْتَ الحِبَّ لَيْلًا"],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:8]: ["صِفْهُ لِي يَا مَنْ رَأَيْتَ الحِبَّ لَيْلاً"],

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+            ["إِنَّ عَيْنِي تَشْتَهِي ذَاكَ الجَمَالَا"],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:9]: ["إِنَّ عَيْنِي تَشْتَهِي ذَاكَ الجَمَالَاً"],

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+
+            ["صِفْ جَمِيلًا كَامِلًا خَلْقًا وَخُلُقًا"],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:11]: ["صِفْ جَمِيلاً كَامِلاً خَلْقاً وَخُلُقاً"],

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+            ["مِنْ جَمَالِ الذَّاتِ قَدْ حَازَ الكَمَالَا"],
+
+            ["صِفْ مَلِيحًا طَرْفُهُ أَسْيَى العَوَالِمْ "],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:14]: ["صِفْ مَلِيحاً طَرْفُهُ أسْبَى العَوَالِمْ "],

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+            ["صِفْ وَجِيهًا وَجْهُهُ حَازَ الكَمَالَا"],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:15]: ["صِفْ وَجِيهاً وَجْهُهُ حَازَ الكَمَالَا"],

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+
+            ["صِفْهُ لِي بِالسُّنْدُسِيَّةِ حِيْنَ يَبْدُو "],
+            ["كَامِلَ الأَوْصَافِ قَدْ مَلَكَ الدَّلَالَا"],
+
+            ["صِفْ عُيُونَ الهَاشِمِي صِفْ لِي المُحَيَّا"],
+            ["صِفْ لِي ثَغْرًا بِابْتِسَامَتِهِ تَلَالًا"],
```
Comment (thenotepadwriter) [inline site/data/qasida/sifhu-li-ya-man-ra-ayta.json:21]: ["صِفْ لِي ثَغْراً بِابْتِسَامَتِهِ تَلَالا"],

**Review by thenotepadwriter — CHANGES_REQUESTED** [review]
Salaams!

I only looked at the Arabic, and made edits to the harakat. Just note that it looks like Sayed Maan uses an automatic tashkeel tool (w Allahu a'lam), and it's not very accurate. 2 tiny tips come to my mind which might be helfpul in future:

- The tanween ً  only ever comes on the last letter of a word (if it's anywhere else, you know it's wrong!) 
- The letter ا (alef) can only ever be preceded with a fath َ so if the letter before it has any other harakah, you can also immediately tell it's not right...

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json (hunk 14 above)
Comment (adehad) [reply site/data/qasida/sifhu-li-ya-man-ra-ayta.json:21]: did this one mean to drop the fatha on the end?
ie would it be
```suggestion
            ["صِفْ لِي ثَغْراً بِابْتِسَامَتِهِ تَلَالَا"],
```

File: site/data/qasida/sifhu-li-ya-man-ra-ayta.json [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+
+            ["صفْ جَمِيلًا أَكْحَلًا مِنْ غَيْرِ كُحْلٍ"],
+            ["أَدْعَجًا عَيْنَاهُ تُنْسِيكَ الغَزَالَا"],
+
+            ["صِفْ غَضُوضَ الطَّرفِ بَسَّامُ المُحَيَّا "],
+            ["أَنْجَلًا تُنْسِيكَ طَلْعَتُهُ الهِلَالَا"],
+
+            ["صِفْ جَمِيلَ القَدِّ وَرْدِيَّ الوِجَانَا "],
+            ["صِفْ مَلِيحًا حُسْنُهُ فَاقَ الخَيَالًا"],
```
[the remark this answers is not in the document]
Comment (adehad) [reply site/data/qasida/sifhu-li-ya-man-ra-ayta.json:30]: also this one?

```suggestion
            ["صِفْ مَلِيحاً حُسْنُهُ فَاقَ الخَيَالَا"],
```

**Review by adehad — COMMENTED** [review]
Wa Salaams wa BarakAllah feekunna!

Sorry I edited your comments so they showed the before and after as I was struggling to see the differences sometimes haha
image

I think there are a few I have some questions about the ending on this, but I can imagine that `لا` is only read in one way so maybe that's why it is less of an issue?

**Comment by thenotepadwriter** [comment]
Wa alaykum assalam, wa fekum barak!

Ah, can you show me how to do those before/after comments in future (from the Android app...)?

I will look at the comments now isA

**Comment by thenotepadwriter** [comment]
I'm sorry I'm struggling to reply to the comments directly (see screenshot), not sure why. But here are my replies:



There are 2 changes here:

ثغراً - The tanween moves from the penultimate letter to the last letter

تلالَا - The tanween is removed from the penultimate letter and the fath remains

There are 2 changes here:

1. مليحاً - The tanween moves to the last letter

2. الخيالَا - The tanween is removed and we keep the fath

**Comment by thenotepadwriter** [comment]
And if this explanation helps at all:

There are 2 different لا endings used in this Qasida, some need a tanween (which is not pronounced when we stop on the word, and would only be pronounced if we joined it with the next word). The other kind does not need a tanween, that's why I added it in some places and removed it in others

**Comment by adehad** [comment]
> And if this explanation helps at all:
> 
> There are 2 different لا endings used in this Qasida, some need a tanween (which is not pronounced when we stop on the word, and would only be pronounced if we joined it with the next word). The other kind does not need a tanween, that's why I added it in some places and removed it in others

this does make a lot of sense thanks

> I'm sorry I'm struggling to reply to the comments directly (see screenshot), not sure why. But here are my replies:
> 
> 

this is a mystery to me as well


> 
> There are 2 changes here:
> 
> ثغراً - The tanween moves from the penultimate letter to the last letter
> 
> تلالَا - The tanween is removed from the penultimate letter and the fath remains
> 
> There are 2 changes here:
> 
>     1. مليحاً - The tanween moves to the last letter
> 
>     2. الخيالَا - The tanween is removed and we keep the fath

perfect



> Ah, can you show me how to do those before/after comments in future (from the Android app...)?
> 

I think it is also the same `±` symbol in the mobile app to look out for

**Comment by thenotepadwriter** [comment]
Khair, jazakum Allah khairan! & bismillah..

**Review by thenotepadwriter — APPROVED** [review]

**Pull request merged**